src/payment/strategies/affirm/affirm-payment-strategy.ts

```diff
@@ -0,0 +1,273 @@
+        const { publicKey } = paymentMethod.initializationData;
+
+        affirmJs(publicKey, this._getScriptURI(testMode));
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const paymentId = payload.payment && payload.payment.methodId;
+        const { useStoreCredit } = payload;
+        const state = this._store.getState();
+        const config = state.config.getStoreConfig();
+
+        if (!paymentId) {
+            throw new PaymentArgumentInvalidError(['payment.methodId']);
+        }
+
+        if (!config) {
+            throw new MissingDataError(MissingDataErrorType.MissingCheckoutConfig);
+        }
+
+        return this._store.dispatch(this._remoteCheckoutActionCreator.initializePayment(paymentId, { useStoreCredit }))
+            .then(affirm.checkout(this._initializeCheckout(useStoreCredit)))
+            .then(affirm.checkout.open())
+            .then(affirm.ui.ready(
+                () => {
+                    affirm.ui.error.on('close', () => {
+                        window.location.href = `${config.links.checkoutLink}.php?action=set_external_checkout&provider=affirm&status=cancelled`;
+                    });
+                }
+            ))
+            .then(() => new Promise<never>(() => {  }));
+
+    }
+
+    deinitialize(options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        if (affirm) {
+            affirm = undefined;
+        }
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    finalize(options: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        return this._store.dispatch(this._remoteCheckoutActionCreator.loadSettings(options.methodId))
+            .then(state => {
+                const payment = state.payment.getPaymentId();
+                const paymentMethod = state.paymentMethods.getPaymentMethod(options.methodId, options.gatewayId);
+                const affirm = state.remoteCheckout.getCheckout('affirm');
+
+                if (!payment) {
+                    throw new MissingDataError(MissingDataErrorType.MissingCheckout);
+                }
+
+                if (!paymentMethod) {
+                    throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+                }
+
+                if (!affirm || !affirm.settings) {
+                    throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+                }
+
+                const orderPayload = {
+                    useStoreCredit: affirm.settings.useStoreCredit,
+                };
+
+                const paymentPayload = {
+                    methodId: payment.providerId,
+                    paymentData: { nonce: paymentMethod.initializationData.token },
+                };
+
+                return this._store.dispatch(this._orderActionCreator.submitOrder(orderPayload, options))
```

Comment:
The disadvantage of going to 3rd party site (like Affirm in this case) with Cart instead of Order is the cart amount could change. It doesn't make sense to clear payment first and then create an order for that order.
Keep in mind that we finalise payment when BigPay sends an asynchronous notification request to BC to confirm the payment has been cleared.
As David mentioned above we should avoid creating these special cases.